src/BaGet.GCP/Services/GoogleCloudStorageService.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Security.Cryptography;
+using System.Threading;
+using System.Threading.Tasks;
+using BaGet.Core.Services;
+using BaGet.GCP.Configuration;
+using Google;
+using Google.Cloud.Storage.V1;
+using Microsoft.Extensions.Options;
+
+namespace BaGet.GCP.Services
+{
+    public class GoogleCloudStorageService : IStorageService
+    {
+        private readonly string _bucketName;
+
+        public GoogleCloudStorageService(IOptionsSnapshot<GoogleCloudStorageOptions> options)
+        {
+            _bucketName = options.Value.BucketName;
```

Comment:
Please add `if (options == null) throw new ArgumentNullException(nameof(options));` before storing the `_bucketName`

Reply:
Will do.

BTW, was just copying this existing code; does it need to be fixed? https://github.com/loic-sharma/BaGet/blob/873b133c619e2bc048bbedc5a7fa8007c688eaf8/src/BaGet.AWS/S3StorageService.cs#L22

Reply:
Maybe never mind; that's `IOptions<>` vs `IOptionsSnapshot<>` and I don't know the difference (specifically in use and nullability).

Reply:
Good catch. Yes, both should have null check and both should use `IOptionsSnapshot`. Feel free to fix it in this pull request if you'd like. If not, I can take care of it later

Reply:
I can push a new commit to add that check.